lib/spack/spack/test/variant.py

```diff
@@ -638,11 +638,11 @@ def test_satisfies_and_constrain(self):
         b["foobar"] = SingleValuedVariant("foobar", "fee")
         b["shared"] = BoolValuedVariant("shared", True)
 
-        assert not a.satisfies(b)
-        assert b.satisfies(a)
+        assert a.intersects(b)
+        assert b.intersects(a)
```

Comment:
This is because of commutativity of `intersects`. Before the semantic imo was not that clear.

Reply:
Looks correct